### DISCUSSION TITLE
Mqtt and domoticz

### DISCUSSION THREAD

**Discussion by Brugje** [emsesp/EMS-ESP32#2156] [General]
After updating the firmware to 3.7.0 I get these errors in domoticz and only the data for the thermostat is updating.

![image](https://github.com/user-attachments/assets/8f37543a-5282-4b99-a65f-61dba3e78c2c)

I have these 4 but the only one updating is the thermostat (in time) its data is always 0.
![image](https://github.com/user-attachments/assets/5f554be1-9f9e-4014-a393-5a5f25c4c38e)

I have updated to 3.7.1.dev1 but still the same. In Mqtt explorer it looks like all the data is renewed but Domoticz can't read.

Please help
Jan

**Comment by proddy** [comment]
Can you share your system info. Need to see which domoticz setting you are using.

**Comment by Brugje** [comment]
i started 3 weeks ago with a board S3 from BBQKees. It came with version 3.6.5 the stable version at that moment. Then i did a upgrade to version 3.7.0 stable and then it went wrong. All worked with 3.6.5 and domoticz. Replaced firmware to tte newest dev. release but the same problem in domoticz. Changing back to 3.6.5 did not help. 
Domoticz version 2024-6 with settings
![image](https://github.com/user-attachments/assets/a89bcf0d-f75a-4184-a3f6-f828dff918b5)
Brooker is now mosquito 2.0.20 (also updated last week after the problems began)
Do you need more information? 
In the screenshot the gateway is not active because of the errors. When i turn it on it discovers all the switches but no updates.
Groet Jan

**Comment by Brugje** [comment]
Mqtt settings
![image](https://github.com/user-attachments/assets/8e4b2562-b296-430f-ba69-002199347c87)

**Comment by proddy** [comment]
The problem is, I don't use or know Domoticz, so I can't see what is wrong. I expect Domoticz is not fully adapting the MQTT Discovery protocol. You may get help from other people who use Domoticz to compare setups. @Domoberry seems quite knowledgeable and has been talking to the Domoticz people - see https://github.com/emsesp/EMS-ESP32/discussions/2138#discussioncomment-11073073 

If you're comfortable with coding/hacking, you can try making it compatible yourself by using MQTT Explorer and manually updating the config topics until it works. That would be the easiest way, I think.

**Comment by bbqkees** [comment]
I think it's likely the latest EMS-ESP uses some HA-defined formatting that the Domoticz Auto-Discovery is not parsing.

I haven't ran the later Domoticz versions at home lately, will have to check it.

**Reply by proddy** [reply]
I'll do a quick change to remove *all* the template conditions if the EMS-ESP discovery type is Domoticz. There are two Domoticz settings, the first one replaces Number with Sensor.

**Comment by Domoberry** [comment]
Appreciate being tagged in this post, but "quite knowledgeable" is really incorrect. I'm an average Domoticz user who recently upgraded to the E32V2 gateway for my new heat pump. I subscribe to the challenges experienced by @Brugje, I noted the same thing next to a certain new Bosch device not fully supported yet. See [https://github.com/emsesp/EMS-ESP32/discussions/2138](url).
I have also posted on the Domoticz forum and learned that there is likely a discrepancy in the way mqtt-ad (config message) are currently handled in Domoticz. Hopefully experts from EMS-ESP and Domoticz can have some virtual coffee together!

**Comment by Brugje** [comment]
Today I have updated the firmware to 3.7.1 dev2 and all the error messages are gone. No single error any more. 
In domoticz all values show ?? instead of % or C. 
![image](https://github.com/user-attachments/assets/bba89d8e-b9a9-4ae5-95b5-e83b306c264c)
But this is also mentioned in #2138. I follow also this discusion because I have also the MH210

**Reply by proddy** [reply]
that is because i removed the unit of measure for domoticz as I though it wasn't supported. I'll add it back, if you wouldn't mind trying it out. dev-3 coming....

**Comment by Brugje** [comment]
Installed dev-3
Most sensors have there units back only a few not, see screenshot.
![image](https://github.com/user-attachments/assets/49700b51-a9f6-4ae8-beb0-adb281a9e833)
 
A big thanks for your qiuick support. I have some other issues i will ask in #2138

**Reply by proddy** [reply]
Those are basic sensors without a uom (unit of measure) defined, as Home Assistant doesn't need it. The list they use is in https://github.com/home-assistant/core/blob/d7ac4bd65379e11461c7ce0893d3533d8d8b8cbf/homeassistant/const.py#L384

Can you recommend what it should be for Domoticz?

**Comment by Brugje** [comment]
It is a number but domoticz wants an ax-label for it. For the Netherlands I would like to use "x". Stands for the number of times it has occurred. I don't know how this is called in Engish. Maybe someone has a better suggestion.
Greetz Jan

**Reply by proddy** [reply]
could you do a test, use MQTT Explorer and manually add the uom and see if it works? Then I can implement it.

**Reply by Domoberry** [reply]
Just for the record, I noted one device that in Domoticz had "??" as unit of measure (uom).
The related EMS-ESP mqtt sensor config message did not have the line "unit_of_meas" : "......",
Indeed, if you add such line and publish the updated config message using an mqtt tool, an additional Domoticz device is created.
E.g. I noted that for example with "unit_of_meas" : "C" a new temperature device is created in Domoticz.
The issue mentioned below holds a reference to other uom's are supported by Domoticz.

**Comment by proddy** [comment]
Let's move this discussion to this issue: https://github.com/emsesp/EMS-ESP32/issues/2177
